SCSSRefactor

EmmaluFox opened the pull request:
-Added README instructions to prevent SCSS conflicts
-Began creating generic mixins to control SCSS around app.

EmmaluFox committed:
Began creating generic mixins to control SCSS around app.

EmmaluFox committed:
-Added README instructions to prevent SCSS conflicts
Began creating generic mixins to control SCSS around app.

EmmaluFox committed:
-Added README instructions to prevent SCSS conflicts
Began creating generic mixins to control SCSS around app.

EmmaluFox committed:
-Added README instructions to prevent SCSS conflicts
Began creating generic mixins to control SCSS around app.

MichaelWalker commented on README.md:
```diff
@@ -29,3 +25,8 @@ To set this up:
 4. Make sure ".env" tis listed in your .gitignore file.
 5. Access your API key using `process.env.REACT_APP_NASA_API_KEY`
 
+### Styling
+- Please import and use only colors listed under "Site Colors" in the ConstantsMixins.scss file.
+- The same goes for font styles and sizes.
+- This should allow us to easily update the general appearance and color scheme of the app from within ConstantsMixins.scss.
+- Please only style global elements (e.g. div, img, body etc) from within the App.scss file to prevent conflicts.
```
Nice! :)

MichaelWalker commented on src/App.tsx:
```diff
@@ -20,7 +20,6 @@ function App() {
               <header className="header" data-testid="header">
                   <NavBar />
               </header>
-                <HeroImageOfTheDay/>
```
Did we mean to delete this?

MichaelWalker commented on src/Components/CuriosityHero/CuriosityHero.scss:
```diff
@@ -0,0 +1 @@
+//Please read README section on Styling before editing or creating SCSS
```
I probably wouldn't bother putting this on all of our CSS files.
Its going to be a bit annoying adding and reading it at the top of every file! 

I'll just enforce it through code review! :)

MichaelWalker commented on src/Components/DateInputForm/DateInputForm.scss:
```diff
@@ -1,3 +1,5 @@
+//Please read README section on Styling before editing or creating SCSS
+//ToDo: move to App.scss
```
Can we just 'Do' this 'TODO'?

MichaelWalker commented on src/Components/Hero/Hero.scss:
```diff
@@ -1,6 +1,7 @@
-$opportunityThemeColor: grey;
+﻿//Please read README section on Styling before editing or creating SCSS
+
+@import "./src/ConstantsMixins";
```
Does this work? 

I'd expect this to be looking for a file under `src/Components/Hero/src/ConstantsMixins` and then not find anything?

EmmaluFox replied:
seems to work!

MichaelWalker commented on src/App.scss:
```diff
@@ -27,5 +31,81 @@ h1{
+  max-width: 100vw;
+  border-width: 0;
+  margin: 0;
+}
```
I'd be a little suspicious about this.
Is there a reason to think that we'd want all of our images to behave this way by default?

MichaelWalker commented on src/App.scss:
```diff
@@ -27,5 +31,81 @@ h1{
+/* Natural flow and rhythm in articles by default */
+article > * + * {
+  margin-top: 1em;
+}
```
I'm also a little suspicious of this one.

I don't think we are going to have much article content, so its probably fine, but if we did I think we'd want to set custom margins etc for our content rather than having it set globally.

(again unless there is a good reason that we'd want everything to work this way)

MichaelWalker commented on src/Components/HeroImageOfTheDay/HeroImageOfTheDay.scss:
```diff
@@ -0,0 +1,6 @@
+@import "./src/ConstantsMixins";
+
+.imageOfTheDay {
+  @include hero-astronomy-image-of-the-day();
```
I think this is only used here, so its probably easiest for it to not be a mixin (ie, just put the code here)
Mixins are great for when we want to re-use things all over a project.

MichaelWalker commented on src/index.css:
```diff
@@ -1,17 +1,17 @@
 div root{
-  font-size: 13px;  
+  font-size: 13px;
 }
```
I don't think that this works either sadly. I suspect that this is meant to be either

```css
:root {
    font-size: 13px;
}
```
or 
```css
html {
    font-size: 13px;
}
```

would you mind also fixing this while you are here?

MichaelWalker commented on src/ConstantsMixins.scss:
```diff
@@ -1,10 +1,24 @@
+/* Rover Theme Colors */
+$curiosityThemeColor: #a1fea1;
+$spiritThemeColor: #ffa6bc;
+$opportunityThemeColor: #ffeb6b;
```
Could we update this to be a little more consistent on conventions?

In this section we have
- Capitalised
- lowercase-with-hyphens
- camelCase

and also
- RGB
- Hex

as usual, I don't really mind what we pick, but its a bit easier if everything is the same!

MichaelWalker reviewed (changes requested):
Some really good work here!
Just a couple of thoughts. :)

EmmaluFox committed:
Improved coherence to scss ConstantsMixins.scss
Organised font-sizes to prevent conflicts

EmmaluFox committed:
Improved coherence to scss ConstantsMixins.scss
Organised font-sizes to prevent conflicts

EmmaluFox committed:
Added Hero Image back in

EmmaluFox committed:
Made changes from code review:
-Removed annoying notes!
-Did "toDo"

EmmaluFox committed:
Made changes from code review:
-Removed annoying notes!
-Did "toDo"
-Removed global img SCSS

EmmaluFox committed:
Made changes from code review:
-Removed annoying notes!
-Did "toDo"
-Removed global img & article SCSS
-Fixed error in index.css file, renamed to SCSS
-Removed unnecessary mixin for hero image of the day
-Made naming styles and color types more consistent

EmmaluFox committed:
Made changes from code review:
-Removed annoying notes!
-Did "toDo"
-Removed global img & article SCSS
-Fixed error in index.css file, renamed to SCSS
-Removed unnecessary mixin for hero image of the day
-Made naming styles and color types more consistent

EmmaluFox reviewed (commented):
Made changes from code review:
-Added Hero Image of the Day back in
-Removed annoying notes!
-Did "toDo"
-Removed global img & article SCSS
-Fixed error in index.css file, renamed to SCSS
-Removed unnecessary mixin for hero image of the day
-Made naming styles and color types more consistent

MichaelWalker reviewed (approved):
Looks good to me - thanks! Just a couple of merge conflicts to sort out now!

EmmaluFox committed:
Made changes from code review:
-Removed annoying notes!
-Did "toDo"
-Removed global img & article SCSS
-Fixed error in index.css file, renamed to SCSS
-Removed unnecessary mixin for hero image of the day
-Made naming styles and color types more consistent
